src/main/java/de/taimos/pipeline/aws/WithAWSStep.java

```diff
@@ -332,46 +352,36 @@ private void withCredentials(@Nonnull Run<?, ?> run, @Nonnull EnvVars localEnv)
 
 					localEnv.override(AWSClientFactory.AWS_ACCESS_KEY_ID, awsCredentials.getAWSAccessKeyId());
 					localEnv.override(AWSClientFactory.AWS_SECRET_ACCESS_KEY, awsCredentials.getAWSSecretKey());
-
-					this.envVars.overrideAll(localEnv);
 				} else {
 					throw new RuntimeException("Cannot find a Username with password credential with the ID " + this.step.getCredentials());
 				}
+			} else if (!StringUtils.isNullOrEmpty(this.step.getSamlAssertion())) {
+				localEnv.override(AWSClientFactory.AWS_ACCESS_KEY_ID, "access_key_not_used_will_pass_through_SAML_assertion");
```

Comment:
Is there a special reason for this values?

Reply:
Yes, when authenticate with assumeRoleWithSAML, the sts client will still try to use creds (even if none are needed, the AWS SDK is a bit buggy on this point). So I just set some dummy creds.